ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/BlurTransformation.kt

```diff
@@ -0,0 +1,338 @@
+import android.content.Context
+import android.graphics.Bitmap
+import android.os.Build
+import android.renderscript.Allocation
+import android.renderscript.Element
+import android.renderscript.RenderScript
+import android.renderscript.ScriptIntrinsicBlur
+import coil.size.Size
+import coil.transform.Transformation
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlin.math.abs
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+/**
+ * BlurTransformation class applies a Box Blur algorithm on a given Bitmap image.
+ * The blurring is performed by averaging the color values of a square group of
+ * adjacent pixels for each pixel in the image.
+ *
+ * @property radius - The radius of the square used for blurring. Higher values
+ *                    produce a more pronounced blur effect.
+ * @property scale  - The scale factor for resizing the image before blurring.
+ *                    Lower values produce a faster but lower quality blur.
+ */
+internal class BlurTransformation(
+    private val context: Context,
+    private val radius: Float = 25f,
+    private val scale: Float = 0.5f,
+) : Transformation {
+
+    override val cacheKey: String = "${javaClass.name}-$radius"
+
+    override suspend fun transform(
+        input: Bitmap,
+        size: Size,
+    ): Bitmap {
+        return input.blur(context, radius) ?: input
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        return other is BlurTransformation && radius == other.radius
+    }
+
+    override fun hashCode(): Int = radius.hashCode()
+}
+
+internal suspend fun Bitmap.blur(context: Context, radius: Float = 25f): Bitmap {
+    return if (Build.VERSION.SDK_INT >= 31) {
+        this.blur(context = context, radius = radius)
+    } else {
+        // useful for testing
+        val useRenderScript = true
+        if (useRenderScript) {
+            return blurUsingRenderScript(context, radius)
```

Comment:
renderscript is pretty optimized and I believe it can take better advantage of the hardware, so it should be the best option other than going directly to the native API in 31+